iOS/RCTOrientation/Orientation.m

```diff
@@ -20,6 +21,30 @@ + (UIInterfaceOrientationMask)getOrientation {
   return _orientation;
 }
 
+- (void)updateInterfaceOrientation: (UIInterfaceOrientation)orientation
+               withOrientationMask: (UIInterfaceOrientationMask) orientationMask
+{
+  [[NSOperationQueue mainQueue] addOperationWithBlock:^ {
+    if (@available(iOS 16.0, *)) {
+      NSArray<UIScene *> *connectedScenes = [[[UIApplication sharedApplication] connectedScenes] allObjects];
+      if ([connectedScenes count] > 0) {
+        UIWindowScene *windowScene = (UIWindowScene *) connectedScenes[0];
+        [windowScene requestGeometryUpdateWithPreferences: [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:orientationMask]
+                                             errorHandler:^(NSError * _Nonnull error) {
+          NSLog(@"%@",[error localizedDescription]);
```

Comment:
I removed the resolve/reject functionality for 2 reasons.

1) it's likely we will get a lot of rejects since when we rotate from locked portrait to landscape, this error will be triggered based on 1. (https://github.com/discord/react-native-orientation/pull/3/files#r990141366)
2) There was an issue where we were `resolve(nil)` first and then `reject(err)` after. You can't call them both, since the promise state is now fulfilled. `resolve(nil)` was always called and the `reject(err)` would happen asynchronously in the callback causing an error.

I am okay with the NSLogs and I'd rather not add some additional complexity - what do you think @donaldchen

Reply:
Yeah, I'm cool with just the NSLog here. Thanks for the explanation!